docs/ref/checks.txt

```diff
@@ -165,9 +165,11 @@ Model fields
 * **fields.E002**: Field names must not contain ``"__"``.
 * **fields.E003**: ``pk`` is a reserved word that cannot be used as a field
   name.
-* **fields.E004**: ``choices`` must be an iterable (e.g., a list or tuple).
-* **fields.E005**: ``choices`` must be an iterable containing ``(actual value,
-  human readable name)`` tuples.
+* **fields.E004**: ``choices`` must be a dictionary or an iterable
+  (e.g., a list or tuple).
+* **fields.E005**: ``choices`` must be a dictionary that maps actual choice
+  values to human readable names or a sequence of ``(actual value, human readable
+  name)`` tuples.
```

Comment:
This error does match what I get when I trigger it:

```
ticket_31262.Student.year_in_school: (fields.E005) 'choices' must be a dictionary that maps actual choice values to human readable names or a sequence of (actual value, human readable name) tuples.
```

But I think it's not consistent with `E004`, I think we need to choose either "must be a mapping (e.g. a dictionary)" or "must be a dictionary" and used the same in both checks (and ensure both items in the doc match the error msg we chose).

Reply:
Yeah. This is tough. Because we can also support a `django.models.Choices` subclass too. And an iterator. This also vaguely refers to a "sequence", but we've got "iterable" in most places. I don't think it's helpful to try and mention all of this, especially as we normalise everything anyway so that it internally we're only ever seeing a list of 2-tuples. I've come up with something more consistent covering mappings and iterables, but I think that'll be enough for now.